src/utils/__tests__/merge-nodes-with-matching-flavors.spec.ts

```diff
@@ -0,0 +1,162 @@
+import { format } from 'prettier';
+
+import { getCodeFromAst } from '../get-code-from-ast';
+import { getImportNodes } from '../get-import-nodes';
+import { getSortedNodes } from '../get-sorted-nodes';
+
+it('should merge duplicate imports within a given chunk', () => {
+    const code = `
+    import type { A } from 'a';
+    import { Junk } from 'junk-group-1'
+    import type { B } from 'a';
+    import "./side-effects1";
+    // C, E and D will be separated from A, B because side-effects in-between
+    import type { C } from 'a';
+    import { D } from "a";
+    import type { E } from "a";
+    // prettier-ignore
+    import type { NoMerge1 } from "a";
+    // prettier-ignore
+    import { NoMerge2 } from "a";
+    import { H } from 'b';
+    import { F } from 'a';
+    // F Will be alone because prettier-ignore in-between
+
+    import { G } from 'b';
+    import * as J from 'c';
+    import { Junk2 } from 'junk-group-2'
+    import * as K from "c";
+    // * as J, * as K can't merge because both Namespaces
+    import {I} from "c"
+    import { default as Def2 } from 'd';
+    import { default as Def1 } from 'd';
+    import Foo1 from 'e';
+    import Foo2 from 'e';
+    `;
+    const importNodes = getImportNodes(code, { plugins: ['typescript'] });
+
+    const sortedNodes = getSortedNodes(importNodes, {
+        importOrder: [],
+        importOrderBuiltinModulesToTop: false,
+        importOrderCaseInsensitive: false,
+        importOrderGroupNamespaceSpecifiers: false,
+        importOrderMergeDuplicateImports: true,
+        importOrderSeparation: true,
+        importOrderSortSpecifiers: true,
+    });
+    const formatted = getCodeFromAst({
+        nodes: sortedNodes,
+        importNodes,
+        originalCode: code,
+        directives: [],
+    });
+
+    expect(format(formatted, { parser: 'babel' })).toMatchInlineSnapshot(`
+        "import type { A, B } from \\"a\\";
+        import { Junk } from \\"junk-group-1\\";
+
+        import \\"./side-effects1\\";
+
+        // C, E and D will be separated from A, B because side-effects in-between
+        import type { C, E } from \\"a\\";
+        import { D } from \\"a\\";
+
+        // prettier-ignore
+        import type { NoMerge1 } from \\"a\\";
+        // prettier-ignore
+        import { NoMerge2 } from \\"a\\";
+
+        import { F } from \\"a\\";
+        // F Will be alone because prettier-ignore in-between
+        import { G, H } from \\"b\\";
+        import * as J from \\"c\\";
+        import * as K from \\"c\\";
+        // * as J, * as K can't merge because both Namespaces
+        import { I } from \\"c\\";
+        import { default as Def1, default as Def2 } from \\"d\\";
+        import Foo1 from \\"e\\";
+        import Foo2 from \\"e\\";
+        import { Junk2 } from \\"junk-group-2\\";
+        "
+    `);
+});
+it("doesn't merge duplicate imports if option disabled", () => {
```

Comment:
Tests are always appreciated, it's great that you added some. But regarding snapshot tests, there's a special directory for those: https://github.com/IanVS/prettier-plugin-sort-imports/tree/main/tests that uses Jest snapshot testing. Could you move these tests there? That would keep it a bit more consistent.

Reply:
Snapshots  tests are integration tests that check whether the plugin as a whole works as expected. https://github.com/IanVS/prettier-plugin-sort-imports/tree/main/src/utils/__tests__ is meant for unit tests of the single functions in the source code. And while having unit tests won't hurt, personally I agree with you and consider integration tests that check how the plugin works as a whole more important for this plugin.

Reply:
@blutorange I guess I'm confused as to the exact ask.

I'm definitely asserting specific functionality of the merge-nodes-with-matching-flavors, I just found it easier to validate the output by calling through the higher level API, because the output is just source-code, rather than somehow creating "artificial" arrays of nodes and shoving them into just the merge-function itself.

In this next PR: https://github.com/IanVS/prettier-plugin-sort-imports/pull/20 I added a bunch of additional test cases to the same file -- those additional test cases [came from here](https://github.com/IanVS/type-import-codemod/blob/9f3210861a77216f07317537a04ce61467e0df7e/src/transform.test.ts) and those were indeed unit tests as well.

I think one cause of the organizational comments is that I'm using `toMatchInlineSnapshot` for developer-updating convenience, rather than `toEqual` as seen in the source-tests which requires the developer to update any changes by hand.

Options I see:
A. Rewrite tests to target lower level API `mergeNodesWithMatchingImportFlavors` directly
B. Rewrite tests to use `toEqual` as seen in other test cases and the source-tests
C. Rewrite tests to `toMatchInlineSnapshot -> toMatchSnapshot` + to move these tests to the special directory for traditional snapshots

What would you two like me to do?

Reply:
I have now pushed that change of `toMatchInlineSnapshot -> toEqual` to both PRs #19, #20